### PR TITLE
87405: Clean up storage of evidence attachments + records used for validation

### PR DESCRIPTION
**Pull request by dfong-adh**
## Summary

- *This work is behind a feature toggle (flipper): YES/NO* No
This PR is to remove storage of evidence attachment data due to the completion of validating the attachment decryption process.

## Related issue(s)
https://github.com/department-of-veterans-affairs/va.gov-team/issues/87405

## Testing done

- [x] *New code is covered by unit tests*
Unprocessed (encrypted) versions of the evidence attachments will no longer be stored. Hexapdf will always be used for decrypting PDFs in the DR evidence flow.

## What areas of the site does it impact?
This impacts the Decision Review evidence attachment processing

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

**Review by shaunburdick — DISMISSED** [review]
LGTM

**Review by stevenjcumming — CHANGES_REQUESTED** [review]
2 things:

1. Can you refactor the `save_attachment_to_cloud!` so it doesn't disable the Metrics/MethodLength cop. Here's a PR I made you can use: https://github.com/department-of-veterans-affairs/vets-api/pull/17436/files

2. Can you remove the following from line 114 of the spec? It's better to just test the response body.
```ruby
expect(subject).to receive(:render).with(json: form_attachment).and_call_original # rubocop:disable RSpec/SubjectStub
```

File: app/controllers/v0/decision_review_evidences_controller.rb [hunk 1]
```diff
@@ -39,12 +28,26 @@ def save_attachment_to_cloud! # rubocop:disable Metrics/MethodLength
         super
       end
 
-      log_formatted(**common_log_params.merge(is_success: true))
+      params = {
```
Comment (anniebtran) [inline app/controllers/v0/decision_review_evidences_controller.rb]: nit picky, but can we rename this to `log_params`? — when i see `params` in a controller i think of the [`params` hash from the request](https://guides.rubyonrails.org/action_controller_overview.html#parameters)

File: app/controllers/v0/decision_review_evidences_controller.rb [hunk 2]
```diff
@@ -15,38 +15,39 @@ class DecisionReviewEvidencesController < ApplicationController
...
-      common_log_params = {
-        key: :evidence_upload_to_s3,
-        form_id: get_form_id_from_request_headers,
-        user_uuid: current_user.uuid,
-        downstream_system: 'AWS S3',
-        params: {
-          form_attachment_guid:,
-          encrypted: password.present?
-        }
+      params = {
```
Comment (anniebtran) [inline app/controllers/v0/decision_review_evidences_controller.rb:23]: nit picky, but can we rename this to log_params? — when i see params in a controller i think of the [params hash from the request](https://guides.rubyonrails.org/action_controller_overview.html#parameters)

Comment (dfong-adh) [reply app/controllers/v0/decision_review_evidences_controller.rb]: Done 👍

**Comment by dfong-adh** [comment]
> 2 things:
> 
> 1. Can you refactor the `save_attachment_to_cloud!` so it doesn't disable the Metrics/MethodLength cop. Here's a PR I made you can use: https://github.com/department-of-veterans-affairs/vets-api/pull/17436/files
> 2. Can you remove the following from line 114 of the spec? It's better to just test the response body.
> 
> ```ruby
> expect(subject).to receive(:render).with(json: form_attachment).and_call_original # rubocop:disable RSpec/SubjectStub
> ```
@stevenjcumming Thanks for the review, I've updated the PR.

**Review by anniebtran — APPROVED** [review]
LGTM

**Review by stevenjcumming — APPROVED** [review]

**Pull request merged**